packages/oid4vci-demo-frontend/src/pages/SSIInformationRequestPage/index.tsx

```diff
@@ -12,38 +12,59 @@ import {
 
 import '../../css/typography.css'
 import {
-    getCurrentEcosystemGeneralConfig,
-    getCurrentEcosystemPageOrComponentConfig,
-    SSIInformationRequestPageConfig
+  DataFormRow,
+  getCurrentEcosystemGeneralConfig,
+  getCurrentEcosystemPageOrComponentConfig,
+  SSIInformationRequestPageConfig
 } from "../../ecosystem-config";
 import SSIPrimaryButton from "../../components/SSIPrimaryButton";
 import {useLocation, useNavigate} from "react-router-dom";
 import {Buffer} from 'buffer';
 import {useMediaQuery} from "react-responsive";
 import {NonMobile} from "../../index";
+import { extractRequiredKeys, transformDataToObject } from "../../utils/ObjectUtils";
 
-interface Payload {
-    firstName?: string
-    lastName?: string
-    emailAddress?: string
-}
+type Payload = Record<string, string>
 
 type State = {
     data?: any
     isManualIdentification?: boolean
 }
 
+function getInitialState(form: DataFormRow[] | undefined) {
+  if (!form) {
+    return {
+      firstName: '',
+      lastName: '',
+      emailAddress: ''
+    }
+  }
+  return form.flat().reduce((acc, field) => {
+    acc[field.key] = '';
```

Comment:
Why?

Reply:
changed this part of the logic to get the empty object from the ObjectUtils method and then populate it here from the credentialSubject